time-tracker-webapi/test/TimeTracker.Library.Test/Services/AdminReportServiceTest.cs

```diff
@@ -7,9 +14,38 @@ namespace TimeTracker.Library.Test.Services
     public class AdminReportServiceTest
     {
         [Fact]
-        public async Task GetAllUsers_returnsExpectedUsers()
+        public async Task GetAllUsersReport_returnsExpectedUsers_withHours()
```

Comment:
As a general comment on this test I would say that this test combines a bunch of business rules into a single test.
I would try to break this out into several tests like:
- GeneratingAReport_IncludesAllUsers
- GeneratingAReport_AggregatesTimeBasedOnEntryType (or something like that)
...

That way, hopefully, when the business rules change, less effort is required to understand what is going on in the tests.

Reply:
signal vs noise